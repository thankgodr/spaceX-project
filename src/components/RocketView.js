import PropTypes from 'prop-types';
import { Button, Col, Row } from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import { RESERVE_ROCKET, CANCEL_RESERVATION, reserveAction } from '../redux/slice/rockets/rockets';

const RocketView = (props) => {
  const dispatch = useDispatch();
  const acionParam = (id, action) => ({
    type: action,
    id,
  });
  return (
        <Row style={{ marginBottom: '25px' }}>
            <Col md={4}>
                <img src={props.image} className="rocket-image img-fluid" />
            </Col>
            <Col md={8} style={{ textAlign: 'left' }}>
                <h4 className="rocket-title">{props.title}</h4>
                <p>{props.description}</p>
                {
                    !props.reserved && <Button onClick={() => {
                      dispatch(reserveAction(acionParam(props.id, RESERVE_ROCKET)));
                    }
                    }>Reservation</Button>
                }

                {
                    props.reserved && <Button
                        onClick={() => {
                          dispatch(reserveAction(acionParam(props.id, CANCEL_RESERVATION)));
                        }
                        }
                        variant="secondary"
                    >Cancel Reservation</Button>
                }

            </Col>
        </Row>
  );
};

RocketView.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  reserved: PropTypes.bool.isRequired,
};

export default RocketView;