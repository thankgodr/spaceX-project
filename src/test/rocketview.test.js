import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import RocketView from '../components/RocketView';
import store from '../redux/configureStore';

describe('Test Rocketview', () => {
  test('Test rocketview omponent', () => {
    const componnent = renderer.create(
            <Provider store={store}>
                <RocketView image="http://" id={11} reserved={false} description="hhdhd" title="dum" >
                </RocketView>
            </Provider>,

    ).toJSON();
    expect(componnent).toMatchSnapshot();
  });
});