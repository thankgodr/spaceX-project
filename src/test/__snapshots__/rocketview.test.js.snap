// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Rocketview Test rocketview omponent 1`] = `
<div
  className="row"
  style={
    Object {
      "marginBottom": "25px",
    }
  }
>
  <div
    className="col-md-4"
  >
    <img
      className="rocket-image img-fluid"
      src="http://"
    />
  </div>
  <div
    className="col-md-8"
    style={
      Object {
        "textAlign": "left",
      }
    }
  >
    <h4
      className="rocket-title"
    >
       
      dum
    </h4>
    <p>
       
      hhdhd
    </p>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Reservation
    </button>
  </div>
</div>
`;
